bls/src/pod.rs

```diff
@@ -0,0 +1,224 @@
+);
+
+impl Default for SignatureCompressed {
+    fn default() -> Self {
+        Self([0; BLS_SIGNATURE_COMPRESSED_SIZE])
+    }
+}
+
+/// A serialized BLS signature in an affine point representation
+#[serde_with::serde_as]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct Signature(
+    #[serde_as(as = "[_; BLS_SIGNATURE_AFFINE_SIZE]")] pub [u8; BLS_SIGNATURE_AFFINE_SIZE],
+);
+
+impl Default for Signature {
+    fn default() -> Self {
+        Self([0; BLS_SIGNATURE_AFFINE_SIZE])
+    }
+}
+
+/// Size of a BLS proof of possession in a compressed point representation
+pub const BLS_PROOF_OF_POSSESSION_COMPRESSED_SIZE: usize = 96;
+
+/// Size of a BLS proof of possession in an affine point representation
+pub const BLS_PROOF_OF_POSSESSION_AFFINE_SIZE: usize = 192;
+
+/// A serialized BLS signature in a compressed point representation
+#[serde_as]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct ProofOfPossessionCompressed(
+    #[serde_as(as = "[_; BLS_PROOF_OF_POSSESSION_COMPRESSED_SIZE]")]
+    pub  [u8; BLS_PROOF_OF_POSSESSION_COMPRESSED_SIZE],
+);
+
+impl Default for ProofOfPossessionCompressed {
+    fn default() -> Self {
+        Self([0; BLS_PROOF_OF_POSSESSION_COMPRESSED_SIZE])
+    }
+}
+
+/// A serialized BLS signature in an affine point representation
+#[serde_as]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct ProofOfPossession(
+    #[serde_as(as = "[_; BLS_PROOF_OF_POSSESSION_AFFINE_SIZE]")]
+    pub  [u8; BLS_PROOF_OF_POSSESSION_AFFINE_SIZE],
+);
+
+impl Default for ProofOfPossession {
+    fn default() -> Self {
+        Self([0; BLS_PROOF_OF_POSSESSION_AFFINE_SIZE])
+    }
+}
+
+/// Size of a BLS public key in a compressed point representation
+pub const BLS_PUBLIC_KEY_COMPRESSED_SIZE: usize = 48;
+
+/// Size of a BLS public key in an affine point representation
+pub const BLS_PUBLIC_KEY_AFFINE_SIZE: usize = 96;
+
+/// A serialized BLS public key in a compressed point representation
+#[serde_as]
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct PubkeyCompressed(
+    #[serde_as(as = "[_; BLS_PUBLIC_KEY_COMPRESSED_SIZE]")] pub [u8; BLS_PUBLIC_KEY_COMPRESSED_SIZE],
+);
+
+impl Default for PubkeyCompressed {
+    fn default() -> Self {
+        Self([0; BLS_PUBLIC_KEY_COMPRESSED_SIZE])
+    }
+}
+
+/// A serialized BLS public key in an affine point representation
+#[serde_as]
+#[derive(Clone, Copy, Debug, Hash, Eq, PartialEq, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct Pubkey(
+    #[serde_as(as = "[_; BLS_PUBLIC_KEY_AFFINE_SIZE]")] pub [u8; BLS_PUBLIC_KEY_AFFINE_SIZE],
+);
+
+impl Default for Pubkey {
+    fn default() -> Self {
+        Self([0; BLS_PUBLIC_KEY_AFFINE_SIZE])
+    }
+}
+
+// Directly implement the `Pod` and `Zeroable` traits for types that are
+// wrappers around byte arrays.
```

Comment:
nit: maybe this is a bit pedantic, but can you add a clarifying comment saying this is safe *because* they're all byte arrays, and we can't derive it automatically because bytemuck doesn't derive Pod for all sizes of byte arrays?